src/spring/azext_spring/_deployment_factory.py

```diff
@@ -238,20 +248,38 @@ def deployment_settings_options_from_resource(original):
         'instance_count': original.sku.capacity,
         'sku': original.sku,
         'env': original.properties.deployment_settings.environment_variables,
-        'config_file_patterns': _get_origin_config_file_patterns(original.properties.deployment_settings.addon_configs)
+        'config_file_patterns': _get_origin_config_file_patterns(original.properties.deployment_settings.addon_configs),
+        'custom_actuator_port': _get_origin_custom_actuator_port(original.properties.deployment_settings.addon_configs),
+        'custom_actuator_path': _get_origin_custom_actuator_path(original.properties.deployment_settings.addon_configs)
     }
     if original.properties.deployment_settings.container_probe_settings is not None:
         options['disable_probe'] = original.properties.deployment_settings.container_probe_settings.disable_probe
     return options
 
 
 def _get_origin_config_file_patterns(origin_addon_configs):
-    if origin_addon_configs:
+    if APPLICATION_CONFIGURATION_SERVICE_NAME in origin_addon_configs:
         acs_addon = origin_addon_configs.get(APPLICATION_CONFIGURATION_SERVICE_NAME)
         return acs_addon.get(APPLICATION_CONFIGURATION_SERVICE_PROPERTY_PATTERN) if acs_addon is not None else None
     return None
 
 
+def _get_origin_custom_actuator_port(origin_addon_configs):
+    if APP_LIVE_VIEW in origin_addon_configs:
+        alv_addon = origin_addon_configs.get(APP_LIVE_VIEW)
+        if ACTUATOR_PORT in alv_addon:
+            return alv_addon.get(ACTUATOR_PORT)
+    return None
+
+
+def _get_origin_custom_actuator_path(origin_addon_configs):
+    if APP_LIVE_VIEW in origin_addon_configs:
+        alv_addon = origin_addon_configs.get(APP_LIVE_VIEW)
+        if alv_addon.get(ACTUATOR_PATH):
+            return alv_addon.get(ACTUATOR_PATH)
+    return None
+
+
```

Comment:
Can you consider write a general method in util.py that can look up a value from a given dict then all add_on file can leverage it?  Then you can call it like `_get_value_from_dict(origin_addon_config, APP_LIVE_VIEW, ACTUATOR_PATH)`
```python
def _get_value_from_dict(input_dict, *keys):
    for key in keys:
        if not isinstance(input_dict, dict):
            return None
        input_dict = input_dict.get(key)
    return input_dict
```